main/domoticz.cpp

```diff
@@ -468,46 +468,31 @@ void GetAppVersion()
 #if !defined WIN32
 void CheckForOnboardSensors()
 {
-	//Check if we are running on a RaspberryPi
-	std::string sLine = "";
-	std::ifstream infile;
-
-#if defined(__FreeBSD__)
-	infile.open("/compat/linux/proc/cpuinfo");
-#else
-	infile.open("/proc/cpuinfo");
-#endif
-	if (infile.is_open())
-	{
-		while (!infile.eof())
-		{
-			getline(infile, sLine);
-			if (
-				(sLine.find("BCM2708") != std::string::npos) ||
-				(sLine.find("BCM2709") != std::string::npos) ||
-				(sLine.find("BCM2711") != std::string::npos) ||
-				(sLine.find("BCM2835") != std::string::npos) 
-
-				)
-			{
-				_log.Log(LOG_STATUS, "System: Raspberry Pi");
-				
-				//Core temperature of BCM2835 SoC
-				szInternalTemperatureCommand = "/opt/vc/bin/vcgencmd measure_temp";
-				bHasInternalTemperature = true;
-
-				//PI Clock speeds	
-				szInternalARMSpeedCommand = "/opt/vc/bin/vcgencmd measure_clock arm";
-				szInternalV3DSpeedCommand = "/opt/vc/bin/vcgencmd measure_clock v3d";
-				szInternalCoreSpeedCommand = "/opt/vc/bin/vcgencmd measure_clock core";
-				bHasInternalClockSpeeds=true;
-
-				break;
-			}
+	//Check if we have vcgencmd (are running on a RaspberryPi)
+	//
+	int returncode=0;
+	std::vector<std::string> ret = ExecuteCommandAndReturn ("vcgencmd measure_temp",returncode);
```

Comment:
@akamming , here you have the command hard-coded and are not using the defined variables. Could go wrong now if this works but variables are different.